src/Microsoft.ML.TimeSeries/AdaptiveSingularSpectrumSequenceModeler.cs

```diff
@@ -1225,12 +1225,14 @@ internal override void Train(FixedSizeQueue<Single> data)
         /// <param name="data">The training time-series.</param>
         internal override void Train(RoleMappedData data)
         {
-            _host.CheckParam(data != null, nameof(data), "The input series for training cannot be null.");
-            if (data.Schema.Feature.Type != NumberType.Float)
-                throw _host.ExceptUserArg(nameof(data.Schema.Feature.Name), "The feature column has  type '{0}', but must be a float.", data.Schema.Feature.Type);
+            _host.CheckValue(data, nameof(data));
+            _host.CheckParam(data.Schema.Feature.HasValue, nameof(data), "Must have features column.");
+            var featureCol = data.Schema.Feature.Value;
+            if (featureCol.Type != NumberType.Float)
+                throw _host.ExceptParam(nameof(data), "The feature column has  type '{0}', but must be a float.", featureCol.Type);
```

Comment:
>ExceptParam [](start = 28, length = 11)

another case for `ExceptSchemaMismatch`

Reply:
All right. I see this on 1232, but the case of 1229 (could not find *any* column with that role, no matter its name), does not seem to be served by this class of helper. (Which is fine.)